hazelcast-client/src/test/java/com/hazelcast/client/discovery/AwsCloudDiscoveryTest.java

```diff
@@ -119,16 +123,25 @@ public void testAwsAddressProvider() {
                 .setTagKey(AWS_TEST_TAG)
                 .setTagValue(AWS_TEST_TAG_VALUE);
 
+
         ClientConfig clientConfig = new ClientConfig();
-        clientConfig.getNetworkConfig().setAwsConfig(clientAwsConfig);
+        String instanceName = null;
+        if (clientConfig.getInstanceName() != null) {
+            instanceName = clientConfig.getInstanceName();
+        } else {
+            instanceName = "hz.client_testAwsAddressProvider";
+        }
 
-        HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
-        IMap<Object, Object> map = client.getMap("MyMap");
-        map.put(1, 5);
-        assertEquals(5, map.get(1));
+        GroupConfig groupConfig = clientConfig.getGroupConfig();
+        String loggingType = clientConfig.getProperty(GroupProperty.LOGGING_TYPE.getName());
+        LoggingService loggingService = new ClientLoggingService(groupConfig.getName(),
+                loggingType, BuildInfoProvider.getBuildInfo(), instanceName);
 
-        AwsAddressProvider awsAddressProvider = new AwsAddressProvider(clientAwsConfig, client.getLoggingService());
+        AwsAddressProvider awsAddressProvider = new AwsAddressProvider(clientAwsConfig, loggingService);
         Collection<Address> addresses = awsAddressProvider.loadAddresses();
         assertTrue(isNotEmpty(addresses));
+        String instancePrivateIp = System.getenv("HZ_TEST_AWS_INSTANCE_PRIVATE_IP");
+        assumeThat("HZ_TEST_AWS_INSTANCE_PRIVATE_IP is not set", instancePrivateIp, Matchers.<String>notNullValue());
```

Comment:
I'm not sure about this `assumeThat()`. The test passes fine until here, if just `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are set. But if `HZ_TEST_AWS_INSTANCE_PRIVATE_IP` is not set, the whole test will nevertheless be skipped/ignored, although the addresses have been successfully loaded and checked.

This works fine on my machine:
```java
        if (instancePrivateIp != null) {
            assertEquals(instancePrivateIp, addresses.iterator().next().getHost());
        }
```

Reply:
added this check to the beginning similar to the other env. variable checks.

Reply:
Looks good to me. But how do I know what to set for `HZ_TEST_AWS_INSTANCE_PRIVATE_IP`? Setting an env variable looks like it's always the same IP address. Where does this come from?

Reply:
Ah, I found the instance via the `aws-test-tag`. Question is answered.